### PR TITLE
NUTCH-3043 Generator: count URLs rejected by URL filters

### PR DESCRIPTION
**Pull request by sebastian-nagel**
- add counters URL_FILTERS_REJECTED and URL_FILTER_EXCEPTION
- simplify logging statement
- remove unnecessary cast

### CODE REVIEW

File: src/java/org/apache/nutch/crawl/Generator.java [hunk 1]
```diff
@@ -253,10 +256,7 @@ public void map(Text key, CrawlDatum value, Context context)
       try {
         sort = scfilters.generatorSortValue(key, crawlDatum, sort);
       } catch (ScoringFilterException sfe) {
-        if (LOG.isWarnEnabled()) {
-          LOG.warn(
-              "Couldn't filter generatorSortValue for " + key + ": " + sfe);
-        }
+        LOG.warn("Couldn't filter generatorSortValue for " + key + ": " + sfe);
```
Comment (lewismc) [inline src/java/org/apache/nutch/crawl/Generator.java:259]: Please use parameterized logging.
```
LOG.warn("Couldn't filter generatorSortValue for {}: {}”, key, sfe);
```

**Review by lewismc — CHANGES_REQUESTED** [review]
Fantastic @sebastian-nagel 
It would be great if we could augment the [metrics documentation](https://cwiki.apache.org/confluence/display/NUTCH/Metrics) once this is merged.

**Comment by sebastian-nagel** [comment]
Hi @lewismc:
- "use parameterized logging": done
- "augment the [metrics documentation](https://cwiki.apache.org/confluence/display/NUTCH/Metrics) once this is merged.": will do
- "we could also [create a test for the counters](https://cwiki.apache.org/confluence/display/MRUNIT/MRUnit+Tutorial#MRUnitTutorial-TestingCounters).": for now, TestGenerator is not based on MRUNIT. The various Generator::generate(...) return the number of generated segments without a way to access the counters (they're logged, however). I'd prefer to track this in a separate issue, because it would require to many code changes to read the counters.

**Review by lewismc — APPROVED** [review]

**Comment by lewismc** [comment]
Excellent @sebastian-nagel 👍

**Comment by sebastian-nagel** [comment]
Thanks, @lewismc! The metrics wiki page was updated.

**Pull request merged**